cadence/contracts/bridge/FlowEVMBridge.cdc

```diff
@@ -107,23 +117,30 @@ access(all) contract FlowEVMBridge {
     /// @param to: The NFT recipient in FlowEVM
     /// @param tollFee: The fee paid for bridging
     ///
-    access(contract) fun bridgeNFTToEVM(token: @{NonFungibleToken.NFT}, to: EVM.EVMAddress, tollFee: @{FungibleToken.Vault}) {
+    access(all) fun bridgeNFTToEVM(
+        token: @{NonFungibleToken.NFT},
+        to: EVM.EVMAddress,
+        tollFee: @FlowToken.Vault
+    ): @FlowToken.Vault {
         pre {
-            FlowEVMBridgeUtils.validateFee(&tollFee, onboarding: false): "Invalid fee paid"
             !token.isInstance(Type<@{FungibleToken.Vault}>()): "Mixed asset types are not yet supported"
             self.typeRequiresOnboarding(token.getType()) == false: "NFT must first be onboarded"
         }
         let tokenType = token.getType()
-        let tokenID = token.getID()
-        let evmID = CrossVMNFT.getEVMID(from: &token) ?? UInt256(token.getID())
-        // TODO: Enhance metadata handling on briding - URI should provide serialized JSON metadata when requested
-        // Grab the URI from the NFT
+        let tokenID = token.id
+        let evmID = CrossVMNFT.getEVMID(from: &token as &{NonFungibleToken.NFT}) ?? UInt256(token.id)
+        // Grab the URI from the NFT if available
         var uri: String = ""
-        if let display = token.resolveView(Type<MetadataViews.Display>()) as! MetadataViews.Display? {
-            uri = display.thumbnail.uri()
+        if let metadata = token.resolveView(Type<CrossVMNFT.EVMBridgedMetadata>()) as! CrossVMNFT.EVMBridgedMetadata? {
```

Comment:
@nvdtf @aishairzay following up on previous convo - curious what you all think about the resolution and contents of this view

Reply:
Good point. Though I may have to make the `baseURI` value optional for the reverse case since there is no guarantee an ERC721 implements the method (included in extensions [`ERC721Metadata`](https://docs.openzeppelin.com/contracts/2.x/api/token/erc721#ERC721Metadata), [`ERC721MetadataMintable`](https://docs.openzeppelin.com/contracts/2.x/api/token/erc721#ERC721MetadataMintable), and [[`ERC721Full`](https://docs.openzeppelin.com/contracts/2.x/api/token/erc721#ERC721Full)). Do you see that being an issue?

Reply:
I updated `CrossVMNFT.URI` to include `baseURI: String?`. Take a look and lmk if that makes sense given what you had in mind